### PR TITLE
Remove allocation of Some from CompressionTable

### PR DESCRIPTION
**Pull request by patriknw**
* JFR TLAB allocations showed that Some was allocated by the CompressionTable
  in the hot path of outbound stream
* CompressionTable.compress is called several times for each message
* This doesn't matter for the tcp transport since there are so many
  other allocations, but for aeron-udp it's nice to remove since it's
  pretty much the only allocation in the outbound stream

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala [hunk 1]
```diff
@@ -7,41 +7,52 @@ package akka.remote.artery.compress
...
  * @param version Either -1 for disabled or a version between 0 and 127
  */
-private[remote] final case class CompressionTable[T](originUid: Long, version: Byte, dictionary: Map[T, Int]) {
-  import CompressionTable.NotCompressedId
+private[remote] final class CompressionTable[T](
+    val originUid: Long,
+    val version: Byte,
+    _dictionary: Object2IntHashMap[T]) {
+
+  def dictionary: Map[T, Int] = {
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala]: This is used from deserialization of CompressionTable, but that is for the periodic advertisements so not very often. Same for the CompressionTable.apply.

File: akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala [hunk 2]
```diff
@@ -7,41 +7,52 @@ package akka.remote.artery.compress
...
  * INTERNAL API: Versioned compression table to be advertised between systems
  *
  * @param version Either -1 for disabled or a version between 0 and 127
  */
-private[remote] final case class CompressionTable[T](originUid: Long, version: Byte, dictionary: Map[T, Int]) {
-  import CompressionTable.NotCompressedId
+private[remote] final class CompressionTable[T](
+    val originUid: Long,
+    val version: Byte,
+    _dictionary: Object2IntHashMap[T]) {
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala]: it's mutable so not exposing to the outside

Comment (hepin1989) [inline akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala]: Cool

**Review by johanandren — COMMENTED** [review]
Looks like it needs a new equals or changed test to pass.

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala (hunk 1 above)
Comment (raboof) [reply akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala]: And also `invert` :+1:

**Review by raboof — APPROVED** [review]
cool

File: akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala (hunk 1 above)
Comment (patriknw) [reply akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala]: yes, invert is probably very slow, and we have some todo there, but not on hot path

**Pull request merged**